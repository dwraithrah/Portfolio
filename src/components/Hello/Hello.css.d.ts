export const Hello: string;
export const hello: string;
