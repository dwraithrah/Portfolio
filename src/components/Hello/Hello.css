.Hello {
  flex: auto;   
  text-align: center;
  color: rgb(86, 144, 211);
}

.Hello button {
  margin-right: 5px;
  display: inline-block;
}
