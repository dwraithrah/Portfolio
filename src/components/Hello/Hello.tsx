import * as React from 'react';
import * as styles from './Hello.css';
import { getExclamationMarks } from '../../util/';
import Grid from 'material-ui/Grid';
import Button from 'material-ui/Button';
import Typography from 'material-ui/Typography';
import blue from 'material-ui/colors/blue';

export interface HelloProps {
  name: String;
  enthusiasmLevel?: number;
  increment?: () => void;
  decrement?: () => void;
}

export function hello({name, enthusiasmLevel = 1, increment, decrement }: HelloProps) {
  if (enthusiasmLevel <= 0) {
    throw new Error('You could be a little more enthusiastic. :D');
  }
  return (
    <div className={styles.hello}>
      <Grid alignItems="center" container={true}  wrap="wrap">
        <Grid item={true} xs={12} className="greeting">
          <Typography 
            className="greeting"
            type="display2"
            gutterBottom={true}
            align="center"
            style={{color: blue[100]}}
          >
            Hello {name + getExclamationMarks(enthusiasmLevel)}
          </Typography>
        </Grid>
      </Grid>
      <Grid alignItems="center" justify="center" spacing={40} container={true} >
        <Grid item={true} xs={12} sm={6}>
          <Button raised={true} color="primary"onClick={increment}>Increase Happiness</Button>
        </Grid>
        <Grid item={true} xs={12} sm={6}>
          <Button raised={true} color="primary"onClick={decrement}>Decrease Happiness</Button>
        </Grid>
      </Grid>   
    </div>
  );
}

export default hello;
