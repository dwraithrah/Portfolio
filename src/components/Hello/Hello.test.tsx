import * as React from 'react';
import Hello from './Hello';
import { shallow } from 'enzyme';

describe('<Hello />', () => {

  let incHand = jest.fn();
  let decHand = jest.fn();
  let hello = shallow(
    <Hello name="Jimmy" enthusiasmLevel={1} increment={incHand} decrement={decHand} />
  );
  it('renders without crashing', () => {
    expect(hello).toMatchSnapshot();
  });

  it('renders the correct text when no enthusiasm level `1` is given', () => {
    hello = shallow(
    <Hello name="Jimmy" enthusiasmLevel={1} increment={incHand} decrement={decHand}/>
  );
    expect(hello.find('.greeting').text()).toEqual('Hello Jimmy!');
  });

  it('throws when the enthusiasm level is 0', () => {
    expect(() => shallow(
    <Hello name="Jimmy" enthusiasmLevel={0} increment={incHand} decrement={decHand}/>
  )).toThrow();
  });

  it('throws when the enthusiasm level is negatiove', () => {
    expect (() => shallow(
    <Hello name="Jimmy" enthusiasmLevel={-1} increment={incHand} decrement={decHand}/>
  )).toThrow();
  });

});
