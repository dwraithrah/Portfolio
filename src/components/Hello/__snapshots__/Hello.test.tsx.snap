// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <hello
    decrement={[Function]}
    enthusiasmLevel={1}
    increment={[Function]}
    name="Jimmy"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withStyles(GridWrapper)
          alignItems="center"
          container={true}
          wrap="wrap"
>
          <withStyles(GridWrapper)
                    className="greeting"
                    item={true}
                    xs={12}
          >
                    <withStyles(Typography)
                              align="center"
                              className="greeting"
                              gutterBottom={true}
                              style={
                                        Object {
                                                  "color": "#e3f2fd",
                                                }
                              }
                              type="display2"
                    >
                              Hello 
                              Jimmy!
                    </withStyles(Typography)>
          </withStyles(GridWrapper)>
</withStyles(GridWrapper)>,
        <withStyles(GridWrapper)
          alignItems="center"
          container={true}
          justify="center"
          spacing={40}
>
          <withStyles(GridWrapper)
                    item={true}
                    sm={6}
                    xs={12}
          >
                    <withStyles(Button)
                              color="primary"
                              onClick={[Function]}
                              raised={true}
                    >
                              Increase Happiness
                    </withStyles(Button)>
          </withStyles(GridWrapper)>
          <withStyles(GridWrapper)
                    item={true}
                    sm={6}
                    xs={12}
          >
                    <withStyles(Button)
                              color="primary"
                              onClick={[Function]}
                              raised={true}
                    >
                              Decrease Happiness
                    </withStyles(Button)>
          </withStyles(GridWrapper)>
</withStyles(GridWrapper)>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alignItems": "center",
          "children": <withStyles(GridWrapper)
            className="greeting"
            item={true}
            xs={12}
>
            <withStyles(Typography)
                        align="center"
                        className="greeting"
                        gutterBottom={true}
                        style={
                                    Object {
                                                "color": "#e3f2fd",
                                              }
                        }
                        type="display2"
            >
                        Hello 
                        Jimmy!
            </withStyles(Typography)>
</withStyles(GridWrapper)>,
          "container": true,
          "wrap": "wrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withStyles(Typography)
              align="center"
              className="greeting"
              gutterBottom={true}
              style={
                            Object {
                                          "color": "#e3f2fd",
                                        }
              }
              type="display2"
>
              Hello 
              Jimmy!
</withStyles(Typography)>,
            "className": "greeting",
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": Array [
                "Hello ",
                "Jimmy!",
              ],
              "className": "greeting",
              "gutterBottom": true,
              "style": Object {
                "color": "#e3f2fd",
              },
              "type": "display2",
            },
            "ref": null,
            "rendered": Array [
              "Hello ",
              "Jimmy!",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alignItems": "center",
          "children": Array [
            <withStyles(GridWrapper)
              item={true}
              sm={6}
              xs={12}
>
              <withStyles(Button)
                            color="primary"
                            onClick={[Function]}
                            raised={true}
              >
                            Increase Happiness
              </withStyles(Button)>
</withStyles(GridWrapper)>,
            <withStyles(GridWrapper)
              item={true}
              sm={6}
              xs={12}
>
              <withStyles(Button)
                            color="primary"
                            onClick={[Function]}
                            raised={true}
              >
                            Decrease Happiness
              </withStyles(Button)>
</withStyles(GridWrapper)>,
          ],
          "container": true,
          "justify": "center",
          "spacing": 40,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withStyles(Button)
                color="primary"
                onClick={[Function]}
                raised={true}
>
                Increase Happiness
</withStyles(Button)>,
              "item": true,
              "sm": 6,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Increase Happiness",
                "color": "primary",
                "onClick": [Function],
                "raised": true,
              },
              "ref": null,
              "rendered": "Increase Happiness",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withStyles(Button)
                color="primary"
                onClick={[Function]}
                raised={true}
>
                Decrease Happiness
</withStyles(Button)>,
              "item": true,
              "sm": 6,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Decrease Happiness",
                "color": "primary",
                "onClick": [Function],
                "raised": true,
              },
              "ref": null,
              "rendered": "Decrease Happiness",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withStyles(GridWrapper)
            alignItems="center"
            container={true}
            wrap="wrap"
>
            <withStyles(GridWrapper)
                        className="greeting"
                        item={true}
                        xs={12}
            >
                        <withStyles(Typography)
                                    align="center"
                                    className="greeting"
                                    gutterBottom={true}
                                    style={
                                                Object {
                                                            "color": "#e3f2fd",
                                                          }
                                    }
                                    type="display2"
                        >
                                    Hello 
                                    Jimmy!
                        </withStyles(Typography)>
            </withStyles(GridWrapper)>
</withStyles(GridWrapper)>,
          <withStyles(GridWrapper)
            alignItems="center"
            container={true}
            justify="center"
            spacing={40}
>
            <withStyles(GridWrapper)
                        item={true}
                        sm={6}
                        xs={12}
            >
                        <withStyles(Button)
                                    color="primary"
                                    onClick={[Function]}
                                    raised={true}
                        >
                                    Increase Happiness
                        </withStyles(Button)>
            </withStyles(GridWrapper)>
            <withStyles(GridWrapper)
                        item={true}
                        sm={6}
                        xs={12}
            >
                        <withStyles(Button)
                                    color="primary"
                                    onClick={[Function]}
                                    raised={true}
                        >
                                    Decrease Happiness
                        </withStyles(Button)>
            </withStyles(GridWrapper)>
</withStyles(GridWrapper)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "center",
            "children": <withStyles(GridWrapper)
              className="greeting"
              item={true}
              xs={12}
>
              <withStyles(Typography)
                            align="center"
                            className="greeting"
                            gutterBottom={true}
                            style={
                                          Object {
                                                        "color": "#e3f2fd",
                                                      }
                            }
                            type="display2"
              >
                            Hello 
                            Jimmy!
              </withStyles(Typography)>
</withStyles(GridWrapper)>,
            "container": true,
            "wrap": "wrap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withStyles(Typography)
                align="center"
                className="greeting"
                gutterBottom={true}
                style={
                                Object {
                                                "color": "#e3f2fd",
                                              }
                }
                type="display2"
>
                Hello 
                Jimmy!
</withStyles(Typography)>,
              "className": "greeting",
              "item": true,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  "Hello ",
                  "Jimmy!",
                ],
                "className": "greeting",
                "gutterBottom": true,
                "style": Object {
                  "color": "#e3f2fd",
                },
                "type": "display2",
              },
              "ref": null,
              "rendered": Array [
                "Hello ",
                "Jimmy!",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "center",
            "children": Array [
              <withStyles(GridWrapper)
                item={true}
                sm={6}
                xs={12}
>
                <withStyles(Button)
                                color="primary"
                                onClick={[Function]}
                                raised={true}
                >
                                Increase Happiness
                </withStyles(Button)>
</withStyles(GridWrapper)>,
              <withStyles(GridWrapper)
                item={true}
                sm={6}
                xs={12}
>
                <withStyles(Button)
                                color="primary"
                                onClick={[Function]}
                                raised={true}
                >
                                Decrease Happiness
                </withStyles(Button)>
</withStyles(GridWrapper)>,
            ],
            "container": true,
            "justify": "center",
            "spacing": 40,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <withStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  raised={true}
>
                  Increase Happiness
</withStyles(Button)>,
                "item": true,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Increase Happiness",
                  "color": "primary",
                  "onClick": [Function],
                  "raised": true,
                },
                "ref": null,
                "rendered": "Increase Happiness",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <withStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  raised={true}
>
                  Decrease Happiness
</withStyles(Button)>,
                "item": true,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Decrease Happiness",
                  "color": "primary",
                  "onClick": [Function],
                  "raised": true,
                },
                "ref": null,
                "rendered": "Decrease Happiness",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
